virtio-devices/src/block.rs

```diff
@@ -432,10 +434,31 @@ impl Block {
             avail_features |= 1u64 << VIRTIO_BLK_F_RO;
         }
 
+        let topology = disk_image.topology();
+        info!("Disk topology: {:?}", topology);
+
+        let logical_block_size = if topology.logical_block_size > 512 {
+            topology.logical_block_size
+        } else {
+            512
+        };
+
+        // Calculate the exponent that maps physical block to logical block
+        let mut physical_block_exp = 0;
+        let mut size = logical_block_size;
+        while size < topology.physical_block_size {
+            physical_block_exp += 1;
+            size <<= 1;
+        }
+
```

Comment:
The VIRTIO spec didn't make things easy... This introduces complexity for no reason...